commit cfcbfb1382dbac331d8aa92d3a218a16b803b2a9
Author: Josef Bacik <josef@toxicpanda.com>
Date:   Mon May 13 17:21:04 2019 -0700

    mm/filemap.c: enable error injection at add_to_page_cache()
    
    Recently I messed up the error handling in filemap_fault() because of an
    unexpected ENOMEM (related to cgroup memory limits) in add_to_page_cache.
    Enable error injection at this point so I can add a testcase to xfstests
    to verify I don't mess this up again.
    
    [akpm@linux-foundation.org: include linux/error-injection.h]
    Link: http://lkml.kernel.org/r/20190403152604.14008-1-josef@toxicpanda.com
    Signed-off-by: Josef Bacik <josef@toxicpanda.com>
    Reviewed-by: William Kucharski <william.kucharski@oracle.com>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

diff --git a/mm/filemap.c b/mm/filemap.c
index 4157f858a9c6..0e929b4da48b 100644
--- a/mm/filemap.c
+++ b/mm/filemap.c
@@ -24,6 +24,7 @@
 #include <linux/pagemap.h>
 #include <linux/file.h>
 #include <linux/uio.h>
+#include <linux/error-injection.h>
 #include <linux/hash.h>
 #include <linux/writeback.h>
 #include <linux/backing-dev.h>
@@ -882,6 +883,7 @@ static int __add_to_page_cache_locked(struct page *page,
 	put_page(page);
 	return xas_error(&xas);
 }
+ALLOW_ERROR_INJECTION(__add_to_page_cache_locked, ERRNO);
 
 /**
  * add_to_page_cache_locked - add a locked page to the pagecache